### PR TITLE
Client: expose registry errors

### PR DESCRIPTION
**Pull request by NunoAlexandre**
Step 2. of #294 

>  2. Move the CLI TransactionError to the core and have that as the error type of every Message
>     2.1 Adjust the affected tests

### CODE REVIEW

File: core/src/error.rs [hunk 1]
```diff
@@ -19,6 +19,31 @@ use derive_try_from_primitive::TryFromPrimitive;
 use std::convert::{TryFrom, TryInto};
 use thiserror::Error as ThisError;
 
+/// The subset of possible errors having led a transaction to failure.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, ThisError)]
+pub enum TransactionError {
+    #[error(transparent)]
+    RegistryError(RegistryError),
```
Comment (CodeSandwich) [inline core/src/error.rs:26]:
```suggestion
    RegistryError(#[from] RegistryError),
```

File: cli/src/command/org.rs [hunk 2]
```diff
@@ -121,8 +121,7 @@ impl CommandT for Register {
             .await?;
         announce_tx("Registering org...");
 
-        let org_registered = register_org_fut.await?;
-        transaction_applied_ok(&org_registered)?;
+        register_org_fut.await?;
```
Comment (CodeSandwich) [inline cli/src/command/org.rs:124]: Are we checking the transaction `result`?

File: core/src/error.rs [hunk 3]
```diff
@@ -19,6 +19,31 @@ use derive_try_from_primitive::TryFromPrimitive;
...
+            .unwrap_or(TransactionError::OtherDispatchError(dispatch_error))
+    }
+}
+
+impl From<RegistryError> for TransactionError {
+    fn from(registry_error: RegistryError) -> Self {
+        TransactionError::RegistryError(registry_error)
+    }
+}
+
```
Comment (CodeSandwich) [inline core/src/error.rs:46]:
```suggestion
```

File: core/src/error.rs (hunk 1 above)
Comment (NunoAlexandre) [reply core/src/error.rs:26]: Thanks! RegistryError wasn't a `thiserror` before and forgot to do this here when it became
one :v:

File: cli/src/command/org.rs (hunk 2 above)
Comment (NunoAlexandre) [reply cli/src/command/org.rs:124]: great catch! this is just checking the higher level result at the client-level, not at the transaction-level. Will fix. Thanks!

Comment (NunoAlexandre) [reply cli/src/command/org.rs]: :heavy_check_mark:

**Pull request merged**